### PR TITLE
PUB-1373 Crown daily list template

### PR DESCRIPTION
**Pull request by junaidiqbalmoj**
### JIRA link ###

https://tools.hmcts.net/jira/browse/PUB-1373

### Change description ###

Create template for Crown Daily list

**Page Part 1:**
Image 05-10-2022 at 16 00

**Page Part 2:**

Image 10-10-2022 at 07 12

**Page Part 3:**

Image 05-10-2022 at 16 01

**Does this PR introduce a breaking change?** (check one with "x")
[ ] Yes
[x] No

### CODE REVIEW

File: src/main/views/crown-daily-list.njk [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+      {{ courtName }}: {{ headerPart1 }}
+      {% if bill %}
+        <br />/{{ courtName }}: {{ headerPart2 }}
+      {% endif %}
+    </h2>
+    <p class="govuk-body govuk-!-font-weight-bold">{{ listDate }} {{ contentDate }}</p>
+    <p class="govuk-body">{{ listUpdated | replace("DATE", publishedDate) }} {{ publishedTime }}</p>
+    <p class="govuk-body">{{ draft }} {{ versionText }} {{ version }}</p><br/>
+    <p class="govuk-body">
+      {% for line in listData['venue']['venueAddress']['line'] %}
```
Comment (ChrisS1512) [inline src/main/views/crown-daily-list.njk:25]: Also need to include town / county here if present

File: src/main/views/crown-daily-list.njk [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+                      {% if sitting['channel'].length === 0 %}
+                        {% set hearingPlatform = sessionChannel %}
+                      {% else %}
+                        {% for channel in sitting['channel'] %}
+                          {% set hearingPlatform = hearingPlatform + ', ' + channel %}
+                        {% endfor %}
+                        {% set hearingPlatform = hearingPlatform.slice(1) %}
+                      {% endif %}
+                      {% set durationAsHours = 0 %}
+                      {% if sitting['durationAsHours'] !== 0 %}
```
Comment (ChrisS1512) [inline src/main/views/crown-daily-list.njk:137]: This duration section feels better to be done as javascript if possible I think we may have a duration function somewhere in the data manipulation service

**Review by ChrisS1512 — CHANGES_REQUESTED** [review]

File: src/main/service/dataManipulationService.ts [hunk 3]
```diff
@@ -34,6 +34,59 @@ export class DataManipulationService {
     return dailyCauseListData;
   }
 
+  public manipulatedCrownDailyListData(dailyCauseList: string): object {
```
[the remark this answers is not in the document]
Comment (junaidiqbalmoj) [reply src/main/service/dataManipulationService.ts]: As discussed, moved the code to separate file.

File: src/main/views/crown-daily-list.njk (hunk 1 above)
Comment (junaidiqbalmoj) [reply src/main/views/crown-daily-list.njk:25]: done.

File: src/main/views/crown-daily-list.njk (hunk 2 above)
Comment (junaidiqbalmoj) [reply src/main/views/crown-daily-list.njk]: refactor the code and move the code to javascript helper class.

File: src/main/views/crown-daily-list.njk [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+                          {% endif %}
+                          <tr class="govuk-table__row">
+                            <td class="govuk-table__cell no-wrap {{ noBorder }}">{{ sitting['time'] }}</td>
+                            <td class="govuk-table__cell no-wrap {{ noBorder }}">{{ case['caseNumber'] }}</td>
+                            <td class="govuk-table__cell no-wrap {{ noBorder }}">{{ hearing['defendant'] }}</td>
+                            <td class="govuk-table__cell no-wrap {{ noBorder }}">{{ hearing['hearingType'] }}</td>
+                            <td class="govuk-table__cell no-wrap {{ noBorder }}">{{ sitting['formattedDuration'] }} {{ case['caseSequenceIndicator'] }}</td>
+                            <td class="govuk-table__cell no-wrap {{ noBorder }}">{{ hearing['prosecutingAuthority'] }}</td>
+                          </tr>
+                          {% if case['linkedCases'] | length %}
```
Comment (ChrisS1512) [inline src/main/views/crown-daily-list.njk:155]: Just testing this out now. It looks like the horizontal line seperates the linked cases too, where as the example in the ticket only has the horizontal line after the linked cases, so it groups them together. Can this be updated?

Also Linked Cases / Notes are bold here but not in example

**Review by ChrisS1512 — CHANGES_REQUESTED** [review]

**Review by KianKwa — APPROVED** [review]

File: src/main/views/crown-daily-list.njk (hunk 4 above)
Comment (junaidiqbalmoj) [reply src/main/views/crown-daily-list.njk:155]: Good spot :) It is fixed now.

**Review by ChrisS1512 — DISMISSED** [review]
LGTM :)

**Review by KianKwa — DISMISSED** [review]

**Review by ChrisS1512 — APPROVED** [review]

**Review by KianKwa — DISMISSED** [review]

**Review by KianKwa — APPROVED** [review]

**Review by ChrisS1512 — APPROVED** [review]

**Pull request merged**